pkg/tenancy/tenancy.go

```diff
@@ -4,19 +4,26 @@
 package tenancy
 
 import (
+	"context"
 	"net/http"
 	"path"
 
+	"google.golang.org/grpc/metadata"
+
 	"github.com/pkg/errors"
 )
 
+type contextKey int
```

Comment:
Could we model the context key for tenant information after an empty struct instead of integer? The docs of the context package recommend the following about context keys:

> The provided key must be comparable and should not be of type string or any other built-in type to avoid collisions between packages using context

I often see the pattern `type TenantCtxKey struct{}` as efficient and impossible to tamper with. It also doesn't require a constant to be declared and can be used as `context.WithValue(context.Background(), TenantCtxKey{}, "my-tenant)`

Reply:
I followed the guidance from this blog: https://go.dev/blog/context#package-userip

Given we define another type (i.e `contextKey`), doesn't it already cover the "don't use a built-in type"? Is an empty struct "comparable" ?

Reply:
Alright, seems good. I've never personally seem that pattern before.